gr-digital/grc/digital_ofdm_carrier_allocator_cvc.block.yml

```diff
@@ -26,6 +26,12 @@ parameters:
     label: Length tag key
     dtype: string
     default: '"packet_len"'
+-   id: output_is_shifted
+    label: Shift Output
+    dtype: enum
```

Comment:
Can also use `bool` here if you'd like to use a switch button instead of the drop-down menu. 

(In that case lines 33-34 can be removed)

Reply:
Had thought as much but this construct was the first I found & so that's what went with.

I can go either way: enum or bool. Is there an advantage / disadvantage to either? If I recall, in "old" GRC XML, using a bool type one couldn't set it to an arbitrary variable value, while using an enum one could. Maybe the same holds true for "new" GRC YML? I'm not familiar with YML yet.

Reply:
That's an interesting point. You're right, with the bool input widget, you can't instead use a variable name, which is kind of a bummer.

So: 

* `bool` is the "sensible" type here, but
* `enum` gives us the ability to actually use a variable at this point.

Hm. Should we be extending the `bool` entry widget to somehow accept text/variable names somehow? And how would that look preferably?

Reply:
My vote about extending the `bool` entry widget is: YES!!! Or, rather, I'd like to see an option added to -any- type that allows for using a variable; not just `bool` type ... but, I'd guess that most other types allow for it already (as they did in "old" GRC XML) ... so, augmenting `bool` might be just what's needed.

Reply:
@michaelld @marcusmueller I haven't really used the old GRC much, could you explain what you mean by "using a variable"?

Reply:
Using a single variable or parameter (for a hier block) to set multiple block's settings. For example, `verbose` or `packet_length_tag_string`. A variable is used in just the local GRC context, while a parameter is settable in the hier block from a GRC context using the hier block. I'm a big fan of hier blocks and parameters, especially `verbose`, `debug`, `log`, and the like ... all of which are (or can be) `bool` types.

Reply:
Oh right :smiley: I don't think I've seen that used anywhere yet, but it does indeed sound useful!